R/dvhydrograph-data.R

```diff
@@ -103,6 +105,21 @@ parseRefData <- function(data, series) {
   return(plotData)
 }
 
+parseEstimatedEdges <- function(data, allVars){
+  datasets <- unlist(allVars, recursive = FALSE)[which(grepl("stat\\d", names(unlist(allVars, recursive = FALSE))))]
+  times <- datasets[which(grepl("time", names(datasets)))]
+  values <- datasets[which(grepl("value", names(datasets)))]
+  estData <- data.frame(time=unname(unlist(times[which(grepl("est", names(times)))])), value=unname(unlist(values[which(grepl("est", names(values)))])), set="est")
+  statData <- data.frame(time=unname(unlist(times[which(!grepl("est", names(times)))])), value=unname(unlist(values[which(!grepl("est", names(values)))])), set="stat")
```

Comment:
not a huge fan of what's happening here at the beginning - you go through and grab all of the times and values into single lists and then separate back out into est and stat dataframes. I think the more logical way would be to pull the est and stat data from allVars separately:

```r
est_df_list <- lapply(allVars[grepl('est_stat\\d', names(allVars))], as.data.frame, stringsAsFactors=F)
estData <- bind_rows(est_df_list) %>% select(-legend.name) %>% mutate(set=rep('est', nrow(.)))

stat_df_list <- lapply(allVars[grepl('^stat\\d', names(allVars))], as.data.frame, stringsAsFactors=F)
statData <- bind_rows(stat_df_list) %>% select(-legend.name) %>% mutate(set=rep('stat', nrow(.)))
```

then you can continue on to the code below.